### DISCUSSION TITLE
WG SXL Standardization 2024-10-10

### DISCUSSION THREAD

**Discussion by emiltin** [rsmp-nordic/community#103] [Working Groups]
## Participants
Thomas, Martin, David, Emil

## Terminology
Presence, occupancy, saturation, density flow?

## Detector logics vs. physical detector
In TLCs, there might be a many-to-many relation between detector logics and physical sensors.

```mermaid
graph LR
DL1 --> S1
DL1 --> S2
DL2 --> S3
DL3 --> S3
```

In the current TLC SXL, physical sensors are not defined as  RSMP components.

In a sensor, is there a need for detector logics? No, DL is a TLC concept.

## Correspondence with TLC SXL
Until we get SXL modules, we will have a bit of duplication between the TLC and then new Sensor SXL.

## Classification
Countries use different classifications. This is challenging to handle with a fixed set of attributes. But we could use a single attribute with a hash of categories:
```json
"counts": {
  "cars": 50
  "bikes": 40
  "pedestrians": 30
}
```

We should have an attribute that clearly indicate the type of classification, e.g sweden, denmark, etc.

Would it be useful to have different levels of classification? E.g.:
default: bicycle, car, truck, bus
detailed: bicycle, cargobikes, large truck, small truck, bus, double bus, tram...

**Comment by RazeAntoine** [comment]
For information, for classification heres is the criterias used in tolling system in France :

- Véhicule height
- Vehicule weigth
- Number of axles
- Caravan or trailer

Perhaps it would be interesting if it were possible for the sensor to return this information and for the categorization not to be an internal process?
